File: Features/vulnerabilities/pythonInterface/PathTraversal/PathTraversal.py
import sys

def convert_args(a):
    args = {}
    for i in range(0, len(a), 2) :
        name = a[i][1:] if a[i][0] == "-" else a[i]
        args[name] = a[i+1]
    return args

args = convert_args(sys.argv[1:])


if args['M'] == 'GET':
    print("Enter your file you want to read in the /home dir")
    print("format: \"-p path\"")
elif args['M'] == 'POST':
    path = args['p']
    if not path:
        print ("No input")
    else:
        fullPath = "/home" + path
        f = open(fullPath, "r")
        print(f.read())

File: Features/vulnerabilities/javaInterface/src/pathTraversal.java
import java.io.File;
import java.util.HashMap;
import java.util.Objects;
import java.util.Scanner;

class PathTraversal {
    public static void main(String[] argsv) {
        HashMap<String, String> args = new HashMap<>();

        for (int i = 0; i < argsv.length; i = i + 2) {
            String name = argsv[i].substring(1);
            args.put(name, argsv[i + 1]);
        }
        if (Objects.equals(args.get("M"), "GET")) {
            System.out.println("Enter the file in your home folder you want to delete in your home folder!");
            System.out.println("Format: \"-p path\"");
        } else if (Objects.equals(args.get("M"), "POST")) {

            System.out.println("Hello Vulnerable World!"); // Display the string.
            String path = args.get("p");

            if (path.startsWith("/home/client/")) {
                File f = new File(path);
                f.delete();
            }
        }
    }
}
